### PR TITLE
ud3/prettifier exception rule improvements

### PR DESCRIPTION
**Pull request by niezbop**
- Restrict rule start to actual exceptions and prevent it from firing on other log with exceptions (for instance on updating/loading/building custom exception classes).

-  Adds alternative end condition to exception rule when the file that raised the exception is unknown (ThreadAbortException for example).

### CODE REVIEW

File: config/log_rules.json [hunk 1]
```diff
@@ -24,10 +24,10 @@
       },
       "exception": {
         "active": true,
-        "start_pattern": "[eE]xception.*: (?<message>.*)\\n",
-        "end_pattern": "Filename: (?:[\\w/:]+/(?<file>\\w+\\.\\w+))? Line: (?<line>-?\\d+)",
+        "start_pattern": "[eE]xception\\w*: (?<message>.*)\\n",
```
Comment (niezbop) [inline config/log_rules.json]: This will only match Exception throwing now

File: config/log_rules.json [hunk 2]
```diff
@@ -24,10 +24,10 @@
...
+        "end_pattern": "(?:Filename: (?:[\\w/:]+/(?<file>\\w+\\.\\w+))? Line: (?<line>-?\\d+)|(?<fileunknown>filename unknown)>:(?<lineunknown>-?\\d+))",
         "start_message": false,
```
Comment (niezbop) [inline config/log_rules.json]: Add alternative ending for exception thrown from unknown files

**Comment by lacostej** [comment]
Could we add some tests, either unit or integration to start covering for those cases?

**Comment by niezbop** [comment]
I'm not sure this is easily doable. Theses fixes are for edge cases that should occur quite rarely. If we want those to be covered we need an extensive log base for most (if not every) things doable with the Unity Editor and make sure and isolate the rule and challenge the log base, making sure that it only catches what was intended (how would you define that correctly?) and nothing else.

In this precise case, issues appeared when I was
1. Creating .unitypackages for classes with custom exception classes
2. Facing ThreadAbortException which seems to be unable to track a throwing file

**Comment by lacostej** [comment]
If we don't have a set of test to go against, we will have a hard time refactoring the rules file without breaking it.

**Comment by niezbop** [comment]
Agreed. I'm opening an issue for that then as I believe this should be done in a separate PR. Better start collecting logs!

**Comment by lacostej** [comment]
Approved. Don't forget to make a better PR message for the changelog. Thanks!
